### PR TITLE
Predicate transformers: Reconciling `Induction.RecStruct` with `Relation.Unary.PredicateTransformer.PT`

### PR DESCRIPTION
**Pull request by jamesmckinna**
Addresses the first point in https://github.com/agda/agda-stdlib/issues/2083#issue-1889007083 ~~plus some knock-on changes from the additional relaxed `Level`s available from taking `PT`-based definitions in place of the stricter `Pt`-based ones.~~ by taking `PT`-based definitions. 

This PR is intended to be minimally invasive, ahead of a possible overhaul of `Induction.WellFounded`. 

UPDATED: have restored the original `Data.Container.Indexed.FreeMonad` and `Effect.Applicative.Predicate` definitions, and removed the `breaking` label. Now this is (almost!) a trivial refactoring, but still worth doing IMHO. No `CHANGELOG` required.

### CODE REVIEW

**Comment by jamesmckinna** [comment]
More `CHANGELOG` hell.

File: src/Induction.agda [hunk 1]
```diff
@@ -17,45 +17,50 @@ module Induction where
...
-Recursor : ∀ {a ℓ₁ ℓ₂} {A : Set a} → RecStruct A ℓ₁ ℓ₂ → Set _
+Recursor : RecStruct A ℓ₁ ℓ₂ → Set _
 Recursor Rec = ∀ P → Rec P ⊆′ P → Universal P
 
 -- And recursors can be constructed from recursor builders.
 
-build : ∀ {a ℓ₁ ℓ₂} {A : Set a} {Rec : RecStruct A ℓ₁ ℓ₂} →
-        RecursorBuilder Rec →
+build : RecursorBuilder Rec →
         Recursor Rec
```
Comment (MatthewDaggitt) [inline src/Induction.agda:52]: Could move this onto one line?

File: src/Induction.agda [hunk 2]
```diff
@@ -17,45 +17,50 @@ module Induction where
...
-                 Pred A ℓ₁ → RecStruct A ℓ₂ ℓ₃ → Set _
+SubsetRecursor : Pred A ℓ → RecStruct A ℓ₁ ℓ₂ → Set _
 SubsetRecursor Q Rec = ∀ P → Rec P ⊆′ P → Q ⊆′ P
 
-subsetBuild : ∀ {a ℓ₁ ℓ₂ ℓ₃}
-                {A : Set a} {Q : Pred A ℓ₁} {Rec : RecStruct A ℓ₂ ℓ₃} →
-                SubsetRecursorBuilder Q Rec →
-                SubsetRecursor Q Rec
+subsetBuild : SubsetRecursorBuilder Q Rec →
+              SubsetRecursor Q Rec
```
Comment (MatthewDaggitt) [inline src/Induction.agda:65]: Could move this onto one line?

**Review by MatthewDaggitt — CHANGES_REQUESTED** [review]
Otherwise looks good!

**Review by MatthewDaggitt — APPROVED** [review]

**Pull request merged**